content/blog/2025/02/camel410-whatsnew/index.md

```diff
@@ -0,0 +1,188 @@
+---
+title: "Apache Camel 4.10 What's New"
+date: 2025-02-12
+authors: [davsclaus, orpiske, tadayosi]
+categories: ["Releases"]
+preview: Details of what we have done in the Camel 4.10 release.
+---
+
+Apache Camel 4.10 LTS has just been [released](/blog/2025/02/RELEASE-4.10.0/).
+
+This release introduces a set of new features and noticeable improvements that we will cover in this blog post.
+
+## Camel Core
+
+Added `customize` to `RouteBuilder` to make it easier to configure a specific Camel component / dataformat, service
+from a Java lambda style, such as follows:
+
+```java
+@Override
+public void configure() throws Exception {
+    customize(KServeComponent.class, k -> {
+        k.getConfiguration().setTarget("localhost:8888");
+    });
+
+    from("timer:kserve?repeatCount=1")
+        .to("kserve:model/metadata?modelName=myModel")
+        .log("${body}");
+}
+```
+
+This makes it possible for low-code users that want to have a single Java file with the Camel route
+and all the Java based configuration done entirely from the same `configure` method.
+
+## Camel JBang
+
+Using _modeline_ has been deprecated. It is recommended to configure externally in the `application.properties`
+file, and then include this file when running with Camel JBang:
+
+```bash
+$ camel run foo.camel.yaml application.properties
+```
+
+Or use `*` to include all files from the current folder:
+
+```bash
+$ camel run *
+```
+
+We have continued to improve the `run` and `export` to work better and also when using different runtimes.
+
+The `debug` command now supports step in and step over. For example debugging a splitter allows
+now to step over and continue after the entire split is complete, while step in, will step
+inside the splitter (default mode).
+
+The `shell` command has been improved, and can now be used for sub commands such as `log` and `trace` or `--watch` mode,
+and be able to exit logging and still be within the shell. Previously you would quit the shell when pressing `ctrl + c`.
+
+TODO: test-infra
+TODO: upgrade command
+
+### Camel JBang Kubernetes
+
+TODO:
+
+## Camel Kamelet
+
+The error handling of the kamelets has been aligned to be Camel _standard_ and act
+similar to Camel routes. See the migration guide for more details.
+
+You can now also configure the `bridgeErrorHandler` option on a Kamelet.
+
+And it's now also possible to call another Kamelet from within a Kamelet.
```

Comment:
This was long time requested. I think it deserves a bit more of emphasis both in the blog and in the official documentation. I'll try to write a short paragraph about it shortly.

Reply:
I've tried a simple custom kamelet (sink) calling log-sink kamelet. It seems we manage to correctly reference the chain of Kamelets, but there is a runtime error at:
```
Caused by: org.apache.camel.FailedToCreateRouteException: Failed to create route recursive-1 at: >>> To[kamelet://sink?routeId=recursive-1] <<< in route: Route(recursive-1)[From[kamelet://source?routeId=recursive-1... because of Failed to resolve endpoint: kamelet://sink?routeId=recursive-1 due to: Property with key [templateId] not found in properties from text: {{templateId}}
```
I can report the full details to a separate Jira and we can work to the features in next release, but I'd avoid mentioning in the blog post if this feature is not yet complete.

Reply:
No leave it, and if you have an issue then create a JIRA

Reply:
I was about to write a Jira with the error, but it turns out this is now working correctly. I'll be adding a new blog to promote this feature during next week.